e2e/validation.go

```diff
@@ -148,6 +149,12 @@ func commonLiveVMValidators(opts *scenarioRunOpts) []*LiveVMValidator {
 		},
 	}
 	validators = append(validators, leakedSecretsValidators(opts)...)
+
+	// kubeletNodeIPValidator cannot be run on older VHDs with kubelet < 1.29
+	if opts.scenario.VHD.Version != config.VHDUbuntu2204Gen2ContainerdPrivateKubePkg.Version {
+		validators = append(validators, kubeletNodeIPValidator())
+	}
```

Comment:
will this add the nodeIPValidator to all scenarios (besides kubelet 1.29)? Do we want that considering it only previously existed on a subset of scenarios?

Reply:
The existing test where is was enabled seem somewhat random. It feels like a copy paste issue where some people copied from test with and without over time. 

I might be wrong

Reply:
https://github.com/Azure/AgentBaker/pull/4824#discussion_r1733514591